### DISCUSSION TITLE
support a shorter file extension

### DISCUSSION THREAD

**Discussion by massivefermion** [gleam-lang/gleam#1335] [Ideas & suggestions]
I know this is not a huge issue but it's better to have the discussion early in the development rather than late. Anyway, `gleam` is pretty long for a file extension and it would look much better if we could use a shorter one. So I looked up a few extensions in [fileinfo](https://fileinfo.com/) and these are the results:

1- [gl](https://fileinfo.com/extension/gl): apparently an old graphical extension that I couldn't even find the programs to open it with(grasp, probably glpro and aftergrasp). The chance of confusion or error is almost none. But I don't know if it's ok to just hijack it. Is there any rules to this?

2- [glm](https://fileinfo.com/extension/glm): model files used by ghoul game engine by raven software. I'm not sure whether they still use that engine or not but it doesn't matter. Again, the chance of confusion or error is almost none. But maybe using an existing file extension is still not a good idea? I don't know!

3- [gli](https://filext.com/file-extension/GLI): fileinfo didn't recognize this one. But apparently it's used by this engineering design software used for designing ground loop heat exchangers(whatever they are). It's pretty niche so I think this one is safe to use too.

**Comment by tranzystorekk** [comment]
I think conflicts are not a big concern since even Rust's `.rs` extension is reported to be used elsewhere in fileinfo.

To me, the most important thing with an extension is a well founded association in a user's head. I'm not sure how other languages solved this problem, but for gleam, the `gl` prefix might have stronger connotations to OpenGL. So IMO nothing speaks "Gleam" as clearly as `.gleam`

**Reply by timjs** [reply]
I agree. Also note that Swift, for example, uses .swift and not a shorter extension. It has the same number of characters as .gleam.

**Comment by cirqol** [comment]
I like what Chris Lattner and the Modular team did with mojo. You can use .mojo or .🔥 I think .⭐️ as an additional extension would be a nice compromise.

**Reply by Rudxain** [reply]
I agree. We're in 2024, every file-manager and text-editor should support UTF-8 paths. Unix-like kernels are "agnostic" to encoding, and (IIRC) Windows-NT uses UTF-16
